server/app/mutations/grid_services/update.rb

```diff
@@ -28,6 +27,17 @@ def validate
       if self.secrets
         validate_secrets_exist(self.grid_service.grid, self.secrets)
       end
+      if self.grid_service.stateful?
+        if self.volumes_from && self.volumes_from.size > 0
+          add_error(:volumes_from, :invalid, 'Cannot combine stateful & volumes_from')
+        end
+        if self.volumes
+          changed_volumes = self.volumes.select { |v| !self.grid_service.volumes.include?(v) }
+          if changed_volumes.any? { |v| !v.include?(':') }
+            add_error(:volumes, :invalid, 'Adding a non-named volume is not supported to a stateful service')
```

Comment:
minor wording fix: "Adding a non-named volume to a stateful service is not supported"